File: typescript/session2/td/agregation/ZeroInductif.ts
import { NaturelInductif, NaturelInductif_defaut } from "./NaturelInductif";
import { EtatNaturelPur } from "./EtatNaturelPur";
import { heriterInterfacesConcretes } from "../../../heritageMultiple/heritageInterfacesconcretes";

export class ZeroInductif implements NaturelInductif {
    constructor() {}
    
    val(): number {
        return 0;
    }

    estNul(): boolean {
        return true;
    }

    predecesseur(): EtatNaturelPur {
        throw new Error("Pas de prédécesseur.");
    }

    chiffre(i: number): number {
        return 0;
    }

    taille(): number {
        return 1;
    }

    representation() : string {
        return "0";
    }

    estEgal(x : EtatNaturelPur) {
        return x.estNul();
    }

    creerZero : () => EtatNaturelPur;   
    creerSuccesseur : (predecesseur : EtatNaturelPur) => EtatNaturelPur; 
    creerNatAvecValeur : (valeur : number) => EtatNaturelPur; 
    creerNatAvecRepresentation : (repDecimal : string) => EtatNaturelPur; 
}

heriterInterfacesConcretes(ZeroInductif, [NaturelInductif_defaut]);

File: java/cours/src/session2/td/agregation/ZeroInductif.java
package session2.td.agregation;

public class ZeroInductif implements NaturelInductif {

	@Override
	public int val() {
		return 0;
	}

	@Override
	public boolean estNul() {
		return true;
	}

	@Override
	public EtatNaturelPur predecesseur() {
		throw new UnsupportedOperationException("Pas de prédécesseur.");
	}

	@Override
	public int chiffre(int i) {
		return 0;
	}

	@Override
	public int taille() {
		return 1;
	}
	
	@Override
	public String toString() {
		return "0";
	}
	@Override
	public boolean equals(Object obj) {
		if(!(obj instanceof EtatNaturelPur))
			return false;
		EtatNaturelPur x = (EtatNaturelPur)obj;
		return x.estNul();
	}



}
